zulipterminal/ui_tools/boxes.py

```diff
@@ -201,6 +201,46 @@ def stream_header(self) -> Any:
         header.markup = stream_title_markup
         return header
 
+    def top_header_bar(self, message_view: Any) -> Any:
+        if self.message['type'] == 'stream':
+            return message_view.stream_header()
+        else:
+            title_markup = ('header', [
+                ('custom', 'You and '),
+                ('custom', self.recipients_names)
+            ])
+            title = urwid.Text(title_markup)
+            header = urwid.AttrWrap(title, 'bar')
+            header.markup = title_markup
+            return header
+
+    def top_search_bar(self, message_view: Any) -> Any:
+        if self.model.narrow == []:
+            text_to_fill = 'All messages'
+        elif len(self.model.narrow) == 1 and \
+                self.model.narrow[0][1] == 'private':
+            text_to_fill = 'All private messages'
+        elif len(self.model.narrow) == 1 and \
+                self.model.narrow[0][1] == 'starred':
+            text_to_fill = 'Starred messages'
+        elif self.message['type'] == 'stream':
+            bar_color = self.model.stream_dict[self.stream_id]['color']
+            bar_color = 's' + bar_color[:2] + bar_color[3] + bar_color[5]
+            text_to_fill = ('bar', [  # type: ignore
+                (bar_color, ' {} '.format(self.caption))
+            ])
+        else:
+            text_to_fill = 'Private message'
+        title_markup = ('header', [
+            ('custom', 'Conversation focus'),
```

Comment:
I recall suggesting this name, but perhaps we can move whatever text we decide on for this to the title for the frame, so it appears above the box instead?

Reply:
Makes sense. Have moved it above.